### PR TITLE
Add the fetcher functionality and documentation on rtd

### PR DESCRIPTION
**Pull request by mraspaud**
This PR adds a pytroll-fetcher script and functionality to download files received through messages, and documentations pages for trollmoves to be hosted on readthedocs.

 - [x] Tests added  
 - [x] Fully documented

### CODE REVIEW

File: docs/source/fetcher.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+---------------
+
+The pytroll-fetcher script and trollmoves.fetcher library are tools to fetch (download) files or objects from local or
+remote filesystems.
+
+Command line script
+*******************
+
+The pytroll-fetcher script listen to posttroll messages and reacts by downloading files to a given locations. The help
+command of the scripts gives::
```
Comment (adybbroe) [inline docs/source/fetcher.rst:11]:
```suggestion
command of the script gives::
```

File: docs/source/fetcher.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    Thanks for using pytroll-fetcher!
+
+An example config would be for this command line interface would be::
+
+    destination: /home/myuser/downloads/
+    publisher_config:
+        nameservers: false
+        port: 1979
+    subscriber_config:
+        addresses:
```
Comment (adybbroe) [inline docs/source/fetcher.rst:34]: Could we perhaps add another example based on topics here if you have a watcher running on the server that already is sending posttroll messages?

**Review by adybbroe — APPROVED** [review]
Looks really nice to me! Thanks. Only one tiny correction.
But would be good to have some more eyes on it. @TAlonglong or @pnuu ?

**Review by TAlonglong — APPROVED** [review]
I had a quick look

LGTM, but I need to run it and test it a bit before I can see if it is doing what it is supposed to. Then eventually we can iterate on it.

File: docs/source/fetcher.rst (hunk 2 above)
Comment (pnuu) [reply docs/source/fetcher.rst:34]: Or point to Posttroll documentation to see which settings are available.

File: docs/source/fetcher.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        - ipc://some_pipe
```
Comment (pnuu) [inline docs/source/fetcher.rst:35]: Is this "inter-process communication" protocol? I didn't know Posttroll supported other than `tcp://`.

Comment (mraspaud) [reply docs/source/fetcher.rst:35]: Yeah, tcp is hardcoded here and there in posttroll, but when passing addresses directly anything zeromq supports should work

File: docs/source/fetcher.rst (hunk 2 above)
Comment (mraspaud) [reply docs/source/fetcher.rst:34]: Fixed with links to the posttroll documentation

**Review by pnuu — APPROVED** [review]
LGTM

**Pull request merged**